crates/lib3h/src/transport/transport_multiplex/mplex.rs

```diff
@@ -0,0 +1,377 @@
+use crate::transport::{error::*, protocol::*};
+use detach::prelude::*;
+use lib3h_ghost_actor::prelude::*;
+use std::{any::Any, collections::HashMap};
+use url::Url;
+
+use super::{LocalRouteSpec, TransportMultiplexRoute};
+
+enum MplexToInnerContext {
+    AwaitBind(
+        GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>,
+    ),
+    AwaitSend(
+        GhostMessage<RequestToChild, RequestToParent, RequestToChildResponse, TransportError>,
+    ),
+}
+
+type SelfContextEndpoint<Context> = GhostContextEndpoint<
```

Comment:
should we promote this type into `transport/protocol.rs`?

Reply:
done